video/out/wayland_common.c

```diff
@@ -1441,8 +1454,87 @@ static void set_geometry(struct vo_wayland_state *wl)
     wl->vdparams.y1 = vo->dheight / wl->scaling;
 }
 
+#ifdef HAVE_GIO
+const GDBusProxyFlags PROXY_FLAGS = G_DBUS_PROXY_FLAGS_DO_NOT_CONNECT_SIGNALS |
+                                    G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES;
+#endif
+
 static int set_screensaver_inhibitor(struct vo_wayland_state *wl, int state)
 {
+#ifdef HAVE_GIO
```

Comment:
ifdefs in the middle of a function are pretty bad for code readability, especially when the code they guard is this verbose.

Reply:
I have moved it out of the functions into one area, and put in stub functions for when the define is not there. If that still looks a bit noisy then it could be moved to a separate file.